core/controller/sync.ts

```diff
@@ -66,20 +61,24 @@ export class SyncController extends BaseController {
     }
 
     @ON('RESPONSE_BLOCKS')
-    async getBlocks(action: { data: { blocks: Array<Block> }, peer }): Promise<void> {
-        const { data, peer } = action;
-        //    TODO call block service and insert blocks
+    async loadBlocks(action: { data: { blocks: Array<Block> }, peer }): Promise<void> {
+        const { data } = action;
+        await BlockService.loadBlocks(data.blocks);
+        if (!SyncService.consensus) {
```

Comment:
I think the consensus must calculate when we need it. Otherwise, the consensus must recalculate after updating each peer.